### ISSUE TITLE
MathLoadTest_bigdecimal_special GPF at libj9jit29.so

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#12281]
Failure link
------------

From an internal build `Test_openjdk17_j9_special.system_aarch64_linux_testList_1/1/`
```
[2021-03-24T08:36:22.092Z] Running test MathLoadTest_bigdecimal_special_5m_6 ...
[2021-03-24T08:36:22.092Z] ===============================================
[2021-03-24T08:36:22.092Z] MathLoadTest_bigdecimal_special_5m_6 Start Time: Wed Mar 24 04:36:19 2021 Epoch Time (ms): 1616574979698
[2021-03-24T08:36:22.092Z] "/home/jenkins/workspace/Test_openjdk17_j9_special.system_aarch64_linux_testList_1/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:destroyAll; "/home/jenkins/workspace/Test_openjdk17_j9_special.system_aarch64_linux_testList_1/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
[2021-03-24T08:36:22.092Z] JVMSHRC005I No shared class caches available
[2021-03-24T08:36:22.092Z] JVMSHRC005I No shared class caches available
[2021-03-24T08:36:22.092Z] cache cleanup done
[2021-03-24T08:36:22.092Z] variation: Mode121
[2021-03-24T08:36:22.092Z] JVM_OPTIONS:  -Xgcpolicy:optavgpause -Xjit:count=0 -Xnocompressedrefs 

openjdk version "17" 2021-09-14
OpenJDK Runtime Environment AdoptOpenJDK (build 17+14-202103240117)
Eclipse OpenJ9 VM AdoptOpenJDK (build master-8cd58be62, JRE 17 Linux aarch64-64-Bit Compressed References 20210324_21 (JIT enabled, AOT enabled)
OpenJ9   - 8cd58be62
OMR      - 165a4a817
JCL      - 32e5e0fb2e based on jdk-17+14)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?TARGET=MathLoadTest_bigdecimal_special_5m_6&amp;SDK_RESOURCE=upstream&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk17_j9_special.system_aarch64_linux&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=0&amp;EXTRA_DOCKER_ARGS=&amp;VENDOR_TEST_DIRS=&amp;TKG_OWNER_BRANCH=AdoptOpenJDK%3Amaster&amp;RUNTIME_NAME=testList_1&amp;PLATFORM=aarch64_linux&amp;KEEP_REPORTDIR=false&amp;PERSONAL_BUILD=false&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2FAdoptOpenJDK%2Fopenjdk-tests.git&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=&amp;BUILD_IDENTIFIER=&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=true&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;KEEP_WORKSPACE=false&amp;JDK_VERSION=17&amp;USER_CREDENTIALS_ID=&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk&amp;RELEASE_TAG=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;UPSTREAM_JOB_NAME=build-scripts%2Fjobs%2Fjdk%2Fjdk-linux-aarch64-openj9&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse%2Fopenj9.git&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=openj9&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=1&amp;DOCKERIMAGE_TAG=&amp;JDK_IMPL=openj9&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;TKG_SHA=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;BUILD_LIST=system&amp;UPSTREAM_JOB_NUMBER=21&amp;TIME_LIMIT=10&amp;JVM_OPTIONS=&amp;PARALLEL=None)

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2021-03-24T08:41:21.435Z] MLT 04:41:19.058 - Completed 87.6%. Number of tests started=2181 (+148)
[2021-03-24T08:41:25.413Z] STF 04:41:23.371 - Heartbeat: Process MLT is still running
[2021-03-24T08:41:31.528Z] MLT stderr Unhandled exception
[2021-03-24T08:41:31.528Z] MLT stderr Type=Segmentation error vmState=0x00000000
[2021-03-24T08:41:31.528Z] MLT stderr J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
[2021-03-24T08:41:31.528Z] MLT stderr Handler1=0000FFFF890C3DA8 Handler2=0000FFFF890393F0 InaccessibleAddress=0000000000000020
[2021-03-24T08:41:31.528Z] MLT stderr R0=0000FFFEEB88F4D0 R1=0000FFFEEB88F4E0 R2=0000FFFEE405FAF0 R3=0000FFFEE40620D0
[2021-03-24T08:41:31.528Z] MLT stderr R4=0000FFFEF3D8C56C R5=0000000000000000 R6=0000000000000000 R7=0000FFFF88CE67E0
[2021-03-24T08:41:31.528Z] MLT stderr R8=0000FFFEEB7F62B8 R9=0000FFFEEB7F61C4 R10=0000FFFEF3D8C574 R11=0000FFFEEB55A5A4
[2021-03-24T08:41:31.528Z] MLT stderr R12=0000000000000001 R13=0000000000000000 R14=0000000000000031 R15=0000FFFEB13AE510
[2021-03-24T08:41:31.528Z] MLT stderr R16=0000FFFF89000118 R17=0000FFFF89F2CDD8 R18=0000FFFF06297290 R19=0000FFFEEB88F4D0
[2021-03-24T08:41:31.528Z] MLT stderr R20=0000FFFEE405FAF0 R21=0000000000000000 R22=0000FFFEF3D8C56C R23=0000000000000000
[2021-03-24T08:41:31.528Z] MLT stderr R24=0000FFFEF3D8C574 R25=0000000000000000 R26=0000FFFF1757FEA0 R27=0000FFFF19096110
[2021-03-24T08:41:31.528Z] MLT stderr R28=0000FFFF19096000 R29=0000000000000000 R30=0000FFFF88A4D4A4 R31=0000FFFEB13AE220
[2021-03-24T08:41:31.528Z] MLT stderr PC=0000FFFF88A4D4A8 SP=0000FFFEB13AE220 PSTATE=0000000020000000
[2021-03-24T08:41:31.528Z] MLT stderr V0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.528Z] MLT stderr V1 00000000bf800000 (f: 3212836864.000000, d: 1.587352e-314)
[2021-03-24T08:41:31.528Z] MLT stderr V2 00000000bf800000 (f: 3212836864.000000, d: 1.587352e-314)
[2021-03-24T08:41:31.528Z] MLT stderr V3 00000000bf800000 (f: 3212836864.000000, d: 1.587352e-314)
[2021-03-24T08:41:31.528Z] MLT stderr V4 00000000bf800000 (f: 3212836864.000000, d: 1.587352e-314)
[2021-03-24T08:41:31.528Z] MLT stderr V5 0000fffeb13ae430 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.528Z] MLT stderr V6 0000fffeb13ae420 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.528Z] MLT stderr V7 0000fffeb13ae410 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.528Z] MLT stderr V8 3fe62e42fefa39ef (f: 4277811712.000000, d: 6.931472e-01)
[2021-03-24T08:41:31.528Z] MLT stderr V9 40734413509f79fe (f: 1352628736.000000, d: 3.082547e+02)
[2021-03-24T08:41:31.529Z] MLT stderr V10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V16 0000fffeb13ae400 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.529Z] MLT stderr V17 0000fffeb13ae3f0 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.529Z] MLT stderr V18 0000fffeb13ae3e0 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.529Z] MLT stderr V19 0000fffeb13ae3d0 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.529Z] MLT stderr V20 0000fffeb13ae3c0 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.529Z] MLT stderr V21 0000fffeb13ae3b0 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.529Z] MLT stderr V22 0000fffeb13ae3a0 (f: 2973426688.000000, d: 1.390643e-309)
[2021-03-24T08:41:31.529Z] MLT stderr V23 00000000000000f0 (f: 240.000000, d: 1.185758e-321)
[2021-03-24T08:41:31.529Z] MLT stderr V24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V29 400aaaaaaaaaaaab (f: 2863311616.000000, d: 3.333333e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V30 4010000000000000 (f: 0.000000, d: 4.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr V31 4010000000000000 (f: 0.000000, d: 4.000000e+00)
[2021-03-24T08:41:31.529Z] MLT stderr Module=/home/jenkins/workspace/Test_openjdk17_j9_special.system_aarch64_linux_testList_1/openjdkbinary/j2sdk-image/lib/default/libj9jit29.so
[2021-03-24T08:41:31.529Z] MLT stderr Module_base_address=0000FFFF88340000
[2021-03-24T08:41:31.529Z] MLT stderr Target=2_90_20210324_21 (Linux 4.18.0-193.28.1.el7.aarch64)
[2021-03-24T08:41:31.529Z] MLT stderr CPU=aarch64 (8 logical CPUs) (0x1dcbd0000 RAM)
[2021-03-24T08:41:31.529Z] MLT stderr ----------- Stack Backtrace -----------
[2021-03-24T08:41:31.529Z] MLT stderr (0x0000FFFF89036048 [libj9prt29.so+0x26048])
[2021-03-24T08:41:31.529Z] MLT stderr ---------------------------------------
[2021-03-24T08:41:31.529Z] MLT stderr JVMDUMP039I Processing dump event "gpf", detail "" at 2021/03/24 04:41:30 - please wait.
```

**Comment by JasonFengJ9** [comment]
Also seen at JDK11 `Test_openjdk11_j9_special.system_aarch64_linux_testList_0/1/`.

**Comment by JasonFengJ9** [comment]
Also seen at JDK 8 `Test_openjdk8_j9_special.system_aarch64_linux_testList_2/5/`:
```
[2021-03-31T04:15:40.231Z] Running test MathLoadTest_all_special_5m_4 ...
[2021-03-31T04:15:40.231Z] ===============================================
[2021-03-31T04:15:40.231Z] MathLoadTest_all_special_5m_4 Start Time: Wed Mar 31 00:15:39 2021 Epoch Time (ms): 1617164139790
[2021-03-31T04:15:40.231Z] "/home/jenkins/workspace/Test_openjdk8_j9_special.system_aarch64_linux_testList_2/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:destroyAll; "/home/jenkins/workspace/Test_openjdk8_j9_special.system_aarch64_linux_testList_2/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
[2021-03-31T04:15:40.231Z] JVMSHRC005I No shared class caches available
[2021-03-31T04:15:40.231Z] JVMSHRC005I No shared class caches available
[2021-03-31T04:15:40.231Z] cache cleanup done
[2021-03-31T04:15:40.231Z] variation: Mode112
[2021-03-31T04:15:40.231Z] JVM_OPTIONS:  -Xgcpolicy:gencon -Xjit:count=0 -Xnocompressedrefs 
```
Similarly at `MathLoadTest_all_special_5m_1` and `MathLoadTest_bigdecimal_special_5m_4`.

**Comment by knn-k** [comment]
Another failure (NPE) with MathLoadTest_all_special_5m_9 JDK 11: #12337

**Comment by knn-k** [comment]
I reproduced the crash locally using the following build.  It is intermittent, and I ran the test three times to see the crash.
```
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.11+7-202103241346)
Eclipse OpenJ9 VM AdoptOpenJDK (build master-8cd58be62, JRE 11 Linux aarch64-64-Bit Compressed References 20210324_694 (JIT enabled, AOT enabled)
OpenJ9   - 8cd58be62
OMR      - 165a4a817
JCL      - 322b40f539 based on jdk-11.0.11+7)
```

**Comment by knn-k** [comment]
This looks like the same problem as #12128, crashing at the following line in `OMR::CodeCache::replaceTrampoline()` with `entry` (= x21) being NULL:
```
entry->_info._resolved._currentTrampoline = trampoline;
```

**Comment by knn-k** [comment]
#12120 suggests it is a platform-independent problem.

**Comment by knn-k** [comment]
I think this issue can be closed now.  It seems it was fixed by #12415.

**Comment by JasonFengJ9** [comment]
Closing as per comment above.
Note: there wasn't much internal tests recently due to 0.27 release, will re-open if this failure appears again.
